python/Makefile

```diff
@@ -0,0 +1,28 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
I think this Makefile make our life easier, but I guess some of them can even go into pyproject.toml if used with [`taskipy`](https://github.com/illBeRoy/taskipy) for arbitrary task execution? 

```toml
[tool.taskipy.tasks]
lint = """
    task lint_isort &&
    task lint_black
"""
lint_isort = "isort src tests"
lint_black = "black src tests"

coverage = """
coverage run --source=src/ -m pytest tests/ &&
coverage report -m --fail-under=90 &&
coverage html &&
coverage xml
"""
```

It can also be used with poetry directly like 
` poetry lint`
or 
`poetry coverage`

Reply:
I actually tried a bit and proposing it in #4923, please let me know if it makes any sense (can be thought of a overkill)